### PR TITLE
RHIROS #533 - Apply filtering on OS column

### PR DESCRIPTION
**Pull request by apuntamb**

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -55,6 +55,78 @@ export const CUSTOM_FILTERS = {
...
+                    label: 'RHEL 8.0',
+                    value: 'RHEL 8.0'
+                },
+                {
+                    label: 'RHEL 8.1',
+                    value: 'RHEL 8.1'
+                },
+                {
+                    label: 'RHEL 8.2',
+                    value: '8.2'
```
Comment (PreetiW) [inline src/constants.js:114]: @apuntamb I think you missed `RHEL` here.

Comment (PreetiW) [reply src/constants.js:114]: also we need to make sure the value here is same as what backend will accept for OS filter value. 💯

File: src/Routes/RosPage/RosPage.js [hunk 2]
```diff
@@ -294,6 +316,16 @@ class RosPage extends React.Component {
                                             onChange: (_e, values) => this.updateStateFilter(values),
                                             value: this.state.stateFilterValue
                                         }
+                                    },
+                                    {
+                                        label: OSFObject.label,
+                                        type: OSFObject.type,
+                                        value: `checkbox-state`,
```
Comment (PreetiW) [inline src/Routes/RosPage/RosPage.js:323]:
```suggestion
                                        value: `checkbox-os`,
```

File: src/Routes/RosPage/RosPage.js [hunk 3]
```diff
@@ -79,6 +80,14 @@ class RosPage extends React.Component {
...
             });
         }
+
+        this.setState({
+            osFilterValue: [
+                'RHEL 7.0', 'RHEL 7.1', 'RHEL 7.2', 'RHEL 7.3', 'RHEL 7.4', 'RHEL 7.5', 'RHEL 7.6',
+                'RHEL 7.7', 'RHEL 7.8', 'RHEL 7.9', 'RHEL 8.0', 'RHEL 8.1', 'RHEL 8.2', 'RHEL 8.3',
+                'RHEL 8.4', 'RHEL 8.5'
+            ]
+        });
```
Comment (PreetiW) [inline src/Routes/RosPage/RosPage.js:90]: @apuntamb can you please remove this.setState from here as we discussed `processQueryParams` method hands the query params which are part of browser URL and we only need to handle it for state when the user routes from Dashboard links to ROS page.

File: src/constants.js (hunk 1 above)
Comment (apuntamb) [reply src/constants.js:114]: Yeah, I have recently made changes in the [backend](https://github.com/RedHatInsights/ros-backend/pull/205/files#diff-a8cad03f08edde21a37e001d0e4efbfb39b3fe06591c7d7ade036dbf91705925R25) and according to those, the values for all should be similar to `8.2`. Will update the same here.

File: src/Routes/RosPage/RosPage.js [hunk 4]
```diff
@@ -181,9 +198,11 @@ class RosPage extends React.Component {
                 stateFilterValue: activeStateFilters
             });
         }
+        // Fixme : how to delete os filter here?
     }
 
     getActiveFilterConfig = () => {
+        // Fixme : how to define os filter here?
         const activeFilters = this.state.stateFilterValue.map((value)=> ({ name: value }));
```
Comment (PreetiW) [inline src/Routes/RosPage/RosPage.js:206]: @apuntamb as we discussed can you update this code to create an array of object for OSFilter value. This method is responsible for creating the active filters chips on the UI. Also feel free to refactor the variable from `activeFilters` to `stateActiveFilters` 

the final array which we want to return will be:
`            [{
                category: 'State',
                chips: stateActiveFilters
            },
            {
                category: 'Operating Systems',
                chips: osActiveFilters
            }]
  `

You might need to add few conditions to check if there are any OS or State filter values selected and create the above array based on it. 
Happy to do call to discuss further on this.
Also for reference go through link here: https://github.com/RedHatInsights/frontend-components/blob/4c5456bd2b27497fbc8849bf5aad99be1d6fb755/packages/components/doc/primary_toolbar.md

File: src/Routes/RosPage/RosPage.js [hunk 5]
```diff
@@ -181,9 +198,11 @@ class RosPage extends React.Component {
                 stateFilterValue: activeStateFilters
             });
         }
+        // Fixme : how to delete os filter here?
```
Comment (PreetiW) [inline src/Routes/RosPage/RosPage.js:201]: @apuntamb as discussed can you add starter code based on your understanding for modifying the deletion logic? after that we can go through it together and update the code if needed. Let me know what you think.
Also this code gets called whenever user selects `Reset Filters` option or `x` button on the filter chip.

**Review by PreetiW — CHANGES_REQUESTED** [review]
Thank you for creating the PR @apuntamb 🙌🏻 
I have left few code suggestions and inline comments, can you please go through it and update the code accordingly and then we can go through it together and do any further modifications needed. 💯

File: src/constants.js (hunk 1 above)
Comment (PreetiW) [reply src/constants.js:114]: @apuntamb Thank you for updating this 💯

**Review by PreetiW — APPROVED** [review]
Thank you @apuntamb for making the changes and resolving all the comments. All changes looks good to me! 

I tested locally and code is working fine, once the backend changes are merged please test and make any other changes needed in constant value file. 💯 

I will request @kgaikwad or @r14chandra to provide review on this PR.

**Review by kgaikwad — APPROVED** [review]

**Comment by kgaikwad** [comment]
Thank you @apuntamb for your frontend contribution :+1:

**Pull request merged**